test/unit/conftest.py

```diff
@@ -0,0 +1,25 @@
+from collections.abc import Generator
+from unittest import mock
```

Comment:
My inclination is to use pytest.monkeypatch when using pytest, rather than using unittest.mock. I'm not sure if that's a general practice or not, though.

Reply:
Fair - done.